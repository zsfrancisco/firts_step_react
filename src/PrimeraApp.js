import React from 'react';
import PropTypes from "prop-types";

// Functional components

const PrimeraApp = ( {saludo,subtitulo='Soy un subtÃ­tulo'} ) => {
    return (
        <>
            <h1>{ saludo }</h1>
            <p>{ subtitulo }</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
}

export default PrimeraApp;