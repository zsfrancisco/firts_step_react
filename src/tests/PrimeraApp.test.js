import '@testing-library/jest-dom';
// import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />', () => {
    // test('Mostrar el mensaje "Hola, soy Francisco"', () => {
    //     const saludo = 'Hola, soy Francisco';
    //     const { getByText } = render( <PrimeraApp saludo='Hola, soy Francisco' /> );
    //     expect(getByText(saludo)).toBeInTheDocument();
    // });

    test('Debe mostrar <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola, Francisco';
        const wrapper = shallow( <PrimeraApp saludo={ saludo } /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar el subtítulo envíado por los props', () => {
        const saludo = 'Hola, Francisco';
        const subtitulo = 'Hola, Jennifer';
        const wrapper = shallow( <PrimeraApp saludo={ saludo } subtitulo={ subtitulo } /> );
        const textoParrafo = wrapper.find('p').text();
        console.log(textoParrafo);
        expect(subtitulo).toBe(textoParrafo);
    });
    
    
})
