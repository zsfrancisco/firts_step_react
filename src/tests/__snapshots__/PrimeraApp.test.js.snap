// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Francisco
  </h1>
  <p>
    Soy un subt√≠tulo
  </p>
</Fragment>
`;
